src/main/kotlin/com/terraformation/backend/admin/AdminCohortsController.kt

```diff
@@ -0,0 +1,169 @@
+package com.terraformation.backend.admin
+
+import com.terraformation.backend.accelerator.db.CohortNotFoundException
+import com.terraformation.backend.accelerator.db.CohortStore
+import com.terraformation.backend.accelerator.db.ModuleStore
+import com.terraformation.backend.accelerator.model.CohortDepth
+import com.terraformation.backend.accelerator.model.CohortModuleDepth
+import com.terraformation.backend.accelerator.model.CohortModuleModel
+import com.terraformation.backend.api.RequireGlobalRole
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.accelerator.CohortId
+import com.terraformation.backend.db.accelerator.ModuleId
+import com.terraformation.backend.db.default_schema.GlobalRole
+import com.terraformation.backend.log.perClassLogger
+import java.time.LocalDate
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.servlet.mvc.support.RedirectAttributes
+
+@Controller
+@RequestMapping("/admin")
+@RequireGlobalRole(
+    [GlobalRole.SuperAdmin, GlobalRole.AcceleratorAdmin, GlobalRole.TFExpert, GlobalRole.ReadOnly])
+@Validated
+class AdminCohortsController(
+    private val cohortStore: CohortStore,
+    private val moduleStore: ModuleStore,
+) {
+  private val log = perClassLogger()
+
+  @GetMapping("/cohorts")
+  fun cohortsHome(model: Model): String {
+    val cohorts = cohortStore.findAll()
+
+    val canReadCohorts = cohorts.all { currentUser().canReadCohort(it.id) }
```

Comment:
It'd be more consistent with the rest of the admin UI to make this throw an access denied exception if the user doesn't have permission to view the page's contents, at which point the template doesn't need this value.

At some point it might be worth talking about whether we want to make "you visited a page you shouldn't have been able to even see a link to" do something else in the admin UI in general, but in the meantime, may as well have it behave consistently.